### DISCUSSION TITLE
How to override the DeviceInfo.OsVersion field from client app?

### DISCUSSION THREAD

**Discussion by nishchith-cp** [microsoft/cpp_client_telemetry#1092] [Q&A]
Need to set the `DeviceInfo.OsVersion` from the client side. 
Currently, the `DeviceInfo.OsVersion` is being populated within the SDK. We want to set this from our client to support the format `xx.xx.xx`

I don't see any APIs exposed for the same. I tried to override the field using setContext API but it doesn't seem to work. 

Kindly help with the same. Is there any way to override this from the client side?

**Comment by nishchith-cp** [comment]
@lalitb Is there any API we can use to set `DeviceInfo.OsVersion`?

**Comment by lalitb** [comment]
The 1ds core C++ provides ISemanticContext::SetOsVersion() to explicitly set the OS version. But I don't see this implemented for iOS - https://github.com/microsoft/cpp_client_telemetry/blob/974c581854d4f6d72a3f889b5520e0e08ba96d86/wrappers/obj-c/ODWSemanticContext.mm. I faster way would be to contribute the change.
